### PR TITLE
router: fix all uses of Sprintf to build host:port for a URL

### PR DESCRIPTION
**Pull request by stbenjam**
IPv6 IP's need to be bracketed.

### CODE REVIEW

File: test/extended/router/scoped.go [hunk 1]
```diff
@@ -136,7 +137,7 @@ var _ = g.Describe("[sig-network][Feature:Router]", func() {
 			pattern := "%s-%s.myapps.mycompany.com"
 
 			g.By("waiting for the healthz endpoint to respond")
-			healthzURI := fmt.Sprintf("http://%s:1936/healthz", routerIP)
+			healthzURI := fmt.Sprintf("http://%s/healthz", net.JoinHostPort(routerIP,"1936"))
```
Comment (Miciah) [inline test/extended/router/scoped.go:140]: Gofmt will want a space here.  
```suggestion
			healthzURI := fmt.Sprintf("http://%s/healthz", net.JoinHostPort(routerIP, "1936"))
```

File: test/extended/router/weighted.go [hunk 2]
```diff
@@ -70,7 +71,7 @@ var _ = g.Describe("[sig-network][Feature:Router]", func() {
 			routerURL := fmt.Sprintf("http://%s", routerIP)
 
 			g.By("waiting for the healthz endpoint to respond")
-			healthzURI := fmt.Sprintf("http://%s:1936/healthz", routerIP)
+			healthzURI := fmt.Sprintf("http://%s/healthz", net.JoinHostPort(routerIP,"1936"))
```
Comment (Miciah) [inline test/extended/router/weighted.go:74]:
```suggestion
			healthzURI := fmt.Sprintf("http://%s/healthz", net.JoinHostPort(routerIP, "1936"))
```

**Review by Miciah — COMMENTED** [review]
The issue might exist in some vendored code too:
```
vendor/k8s.io/kubernetes/test/e2e/network/topology_hints.go:                    cmd := fmt.Sprintf(`date; for i in $(seq 1 3000); do sleep 1; echo "Date: $(date) Try: ${i}"; curl -q -s --connect-timeout 2 http://%s:80/ ; echo; done`, svc.Name)
vendor/k8s.io/kubernetes/test/e2e/upgrades/apps/cassandra.go:   r, err := http.Get(fmt.Sprintf("http://%s:8080/list", t.ip))
vendor/k8s.io/kubernetes/test/e2e/upgrades/apps/cassandra.go:   r, err := http.PostForm(fmt.Sprintf("http://%s:8080/add", t.ip), val)
vendor/k8s.io/kubernetes/test/e2e/upgrades/apps/etcd.go:        r, err := http.Get(fmt.Sprintf("http://%s:8080/list", t.ip))
vendor/k8s.io/kubernetes/test/e2e/upgrades/apps/etcd.go:        r, err := http.PostForm(fmt.Sprintf("http://%s:8080/add", t.ip), val)
vendor/k8s.io/kubernetes/test/e2e/upgrades/apps/mysql.go:       r, err := http.PostForm(fmt.Sprintf("http://%s:8080/addName", t.ip), val)
vendor/k8s.io/kubernetes/test/e2e/upgrades/apps/mysql.go:       r, err := http.Get(fmt.Sprintf("http://%s:8080/countNames", t.ip))
```
But I guess that needs to be fixed upstream.

File: test/extended/router/weighted.go (hunk 2 above)
Comment (stbenjam) [reply test/extended/router/weighted.go]: Thanks!

**Comment by stbenjam** [comment]
> The issue might exist in some vendored code too:

https://github.com/kubernetes/kubernetes/pull/109344

**Comment by Miciah** [comment]
Thanks!
/lgtm

**Pull request merged**